ignite/metrics/accumulation.py

```diff
@@ -51,6 +51,8 @@ def __init__(
         self.num_examples = None
         self._op = op
 
+        if torch.device(device).type == "xla":
```

Comment:
@n2cholas Maybe we can do this check once inside `Metric.__init__`, what do you think ?

Reply:
I initially had this line in `Metric.__init__`, but noticed that the `ConfusionMatrix` had tests on XLA, so I did it this way instead so I didn't break those tests. I would definitely prefer moving that line to `Metric.__init__`, if it's ok to break the existing `ConfusionMatrix` functionality. What do you think?

Reply:
Sorry for delayed answer. 

> if it's ok to break the existing ConfusionMatrix functionality

I think we can do that. You are talking about this test, right ? 
https://github.com/pytorch/ignite/blob/d192a8f43c63e7da922110474603389b1c0e030c/tests/ignite/metrics/test_confusion_matrix.py#L617

Reply:
Yes, I was talking about that test. Sounds good, I'll work on moving the check to `Metric.__init__`!